src/lib/hooks/useBuyNowPrice.ts

```diff
@@ -0,0 +1,21 @@
+//- React Imports
+import { useQuery } from 'react-query';
+
+//- Hooks Imports
+import { useZnsSdk } from './useZnsSdk';
+
+export const useBuyNowPrice = (domainId: string) => {
+	// SDK
+	const sdk = useZnsSdk();
+
+	// Query
+	return useQuery(
+		['domain-buy-now-price', domainId],
+		async () => domainId && (await sdk.zauction.getBuyNowPrice(domainId)),
+		{
+			retry: false,
+			refetchOnMount: false,
+			refetchOnWindowFocus: false,
+		},
+	);
```

Comment:
Set `enabled: domainId` (or maybe `Boolean(domainId)`) in the options, and remove `domainId &&` from the query.

Reply:
Ah I did implement this yesterday but reverted the change in the end. I didn't think it was required as the query in use is not relying on an async return but maybe I've missed something. I would have thought this would be required in other query hooks that take ```domainId```

Reply:
Updated!